qt/widgets/common/src/PropertyHandler.cpp

```diff
@@ -461,20 +461,32 @@ void PropertyHandler::removeFunction() {
         break;
       }
     }
-    ph->renameChildren();
+    ph->renameChildren(cf);
     m_browser->setFitEnabled(cf->nFunctions() > 0);
   }
 }
 
-void PropertyHandler::renameChildren() const {
+void PropertyHandler::renameChildren(std::shared_ptr<Mantid::API::CompositeFunction> cf) {
```

Comment:
I have just one code change suggestion here. It is generally good practice to limit how much you pass around and have shared_ptr objects as function/method arguments unless it really needs to do something with the shared_ptr. In this case we are just querying the composite function so a function signature like

```
void PropertyHandler::renameChildren(const Mantid::API::CompositeFunction &cf)
```

and it would be called as `ph->renameChildren(*cf);`

would describe to the reader that this method does not modify the composite function object or the pointer that is pointing to it. Other than this the code looks good.